### PR TITLE
WIP: Merging the refactoring and testing improvements to RESQPY into master.

### PR DESCRIPTION
**Pull request by cflynn3**

### CODE REVIEW

File: resqpy/fault.py [hunk 1]
```diff
@@ -20,7 +20,7 @@
 import resqpy.olio.uuid as bu
 import resqpy.olio.write_hdf5 as rwh5
 import resqpy.olio.xml_et as rqet
-import resqpy.organize as rqo
+import resqpy.organize.organize_old as rqo
```
Comment (connortann) [inline resqpy/fault.py:23]: @cflynn3 is this intentional? Doesn't look like a change that is ready to be accepted, but it has already been merged into the integration branch.

File: resqpy/organize/__init__.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+__all__ = [
+    'OrganizationFeature',
+]
```
Comment (connortann) [inline resqpy/organize/__init__.py:3]: This is incomplete

File: resqpy/organize/organize_old.py [hunk 3]
```diff
@@ -82,7 +81,7 @@ def create_xml_has_occurred_during(model, parent_node, hod_pair, tag = 'HasOccur
     model.create_ref_node('ChronoTop', model.title_for_root(chrono_top_root), top_chrono_uuid, root = hod_node)
 
 
-def _alias_for_attribute(attribute_name):
+def alias_for_attribute(attribute_name):
```
Comment (connortann) [inline resqpy/organize/organize_old.py:84]: Should this be part of the public API? It is surely an implementation detail for attributes, hard to see why this should be exposed

File: resqpy/organize/organize_old.py [hunk 4]
```diff
@@ -94,102 +93,101 @@ def fset(self, value):
...
+#             return True
+#         if check_extra_metadata and not equivalent_extra_metadata(self, other):
+#             return False
+#         return self.feature_name == other.feature_name
+#
+#     def create_xml(self, add_as_part = True, originator = None, reuse = True):
+#         """Creates a geobody feature xml node from this geobody feature object."""
+#         if reuse and self.try_reuse():
+#             return self.root  # check for reusable (equivalent) object
+#         return super().create_xml(add_as_part = add_as_part, originator = originator)
```
Comment (connortann) [inline resqpy/organize/organize_old.py:183]: We probably do not want these comments included in our code

File: resqpy/organize/organize_old.py (hunk 3 above)
Comment (cflynn3) [reply resqpy/organize/organize_old.py:84]: This method is already being called from other modules (e.g. well.py), therefore the renaming is correct.

File: resqpy/grid.py [hunk 5]
```diff
@@ -321,8 +319,7 @@ def denaturalized_cell_index(self, c0):
         return (k0, j0, i0)
 
     def denaturalized_cell_indices(self, c0s):
-        """Returns an integer numpy array of shape (..., 3) holding kji0 indices for the cells with given natural
-        indices.
+        """Returns an integer array holding kji0 indices for the cells with given natural indices.
 
         argument:
            c0s: numpy integer array of shape (...,) being natural cell indices (for a flattened array)
```
Comment (connortann) [inline resqpy/grid.py:325]: Think the "3" is missing here
```suggestion
           c0s: numpy integer array of shape (..., 3) being natural cell indices (for a flattened array)
```

File: resqpy/grid.py [hunk 6]
```diff
@@ -2854,7 +2853,7 @@ def unsplit_x_section_points(self, axis, ref_slice0 = 0, plus_face = False, mask
             return points[:, ref_slice0, :, :]
 
     def x_section_points(self, axis, ref_slice0 = 0, plus_face = False, masked = False):
-        """Deprecated: please use unsplit_x_section_points() instead."""
+        """Deprecated: please use `unsplit_x_section_points` instead."""
 
         return self.unsplit_x_section_points(axis, ref_slice0 = ref_slice0, plus_face = plus_face, masked = masked)
```
Comment (connortann) [inline resqpy/grid.py:2858]: Given this func is deprecated, we should raise a DeprecationWarning here

**Pull request merged**